ci/test_lib.py

```diff
@@ -0,0 +1,240 @@
+import glob, os, random, re, shutil, statistics, sys
+from dataclasses import dataclass
+from pathlib import Path
+
+# add parent dir to Python path
+parent_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), os.pardir))
+sys.path.append(parent_dir)
+
+from Compiler.library import print_ln
+from Compiler.compilerLib import Compiler
+
+from mpcstats_lib import MAGIC_NUMBER, read_data
+
+from enum import Enum
+import subprocess
+from typing import NamedTuple
+
+class Succ:
+    def __init__(self, tolerance):
+        self.tolerance = tolerance
+
+class Fail:
+    def __init__(self, pattern):
+        self.pattern = pattern
+
+def gen_computation(
+    player_data,
+    selected_col,
+    func,
+    num_params,
+):
+    def get_col(party_id):
+        data = player_data[party_id]
+        mat = read_data(party_id, len(data), len(data[0]))
+        col = [mat[selected_col][i] for i in range(mat.shape[1])]
+        return col
+
+    if num_params == 1:
+        def computation():
+            party_ids = list(range(num_params))
+            col = get_col(party_ids[0])
+            res = func(col).reveal()
+            print_ln('result: %s', res)
+        return computation
+
+    elif num_params == 2:
+        def computation():
+            party_ids = list(range(num_params))
+            col1 = get_col(party_ids[0])
+            col2 = get_col(party_ids[1])
+            res = func(col1, col2).reveal()
+            print_ln('result: %s', res)
+        return computation
+
+    else:
+        runtime_error(f'# of func params is expected to be 1 or 2, but got {num_params}')
```

Comment:
should this be an exception in python instead?
```suggestion
        raise Exception(f'# of func params is expected to be 1 or 2, but got {num_params}')
```

Reply:
I think`runtime_error` would stop the computation more gracefully than raising an exception. The source code is:

```python
def crash(condition=None):
    """ Crash virtual machine.

    :param condition: crash if true (default: true)

    """
    if isinstance(condition, localint):
        # allow crash on local values
        condition = condition._v
    if condition is None:
        condition = regint(1)
    instructions.crash(regint.conv(condition))

def runtime_error(msg='', *args):
    """ Print an error message and abort the runtime.
    Parameters work as in :py:func:`print_ln` """
    print_str('User exception: ')
    print_ln(msg, *args)
    crash()
```

Reply:
I mixed up and was thinking that this part was inside `computation` :). Fixed.